screens/CheckoutScreen.js

```diff
@@ -181,61 +184,113 @@ export default class CheckoutScreen extends React.Component {
     }
   };
 
+  // Returns index of the first product with a name starting with the given letter in products list.
+  // If no product starting with that letter exists, find the next product.
+  getIndexOfFirstProductAtLetter = letter => {
+    let prodList = this.state.products.filter(product => product.name.charAt(0) === letter);
+    let nextLetter = letter;
+    while (prodList.length === 0) {
+      nextLetter = String.fromCharCode(nextLetter.charCodeAt(0) + 1);
+      prodList = this.state.products.filter(product => product.name.charAt(0) === nextLetter);
+    }
+    return this.state.products.indexOf(prodList[0]);
```

Comment:
Nice!! A couple of things, which I think could make this a little more compact (and handle some edge cases):
- Javascript supports comparison with ASCII values, so for alphabetical ordering you can compare character values with `<` and `>`
- Should account for both first letter and last letter, since it's possible that once we start filtering the product lists to be per-store only (very few products in a store), there will be empty tabs?
- Should use `.toLowercase` or `.toUppercase` depending on what you're passing in for 'firstLetter' and 'lastLetter' when comparing, just for robustness (in case an admin mistakenly changes the name of a product)
```
// all products that start with letters between these
let prodList = this.state.products.filter(product => product.name.charAt(0).toLowerCase() <= startLetter && product.name.charAt(0) > endLetter);
if (prodList.length === 0) {
 // error check, since when we start filtering products there might be none that exist
// not rly sure rn but have to return something else
}
return this.state.products.indexOf(prodList[0]);
```

^ not sure if this actually works but you get the idea!